controllers/provisioning_controller.go

```diff
@@ -234,38 +234,8 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		r.Log.V(1).Info("Adding annotation for CBO to take ownership of metal3 deployment created by MAO")
 	}
 
-	// Proceed with creating or updating the Metal3 deployment
-	updated, err := provisioning.EnsureMetal3Deployment(info, metal3DeploymentSelector)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-	if updated {
-		err = r.Client.Status().Update(context.Background(), baremetalConfig)
-		return ctrl.Result{Requeue: true}, err
-	}
-
-	// Determine the status of the deployment
-	deploymentState, err := provisioning.GetDeploymentState(r.KubeClient.AppsV1(), ComponentNamespace, baremetalConfig)
-	if err != nil {
-		err = r.updateCOStatus(ReasonNotFound, "metal3 deployment inaccessible", "")
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("unable to put %q ClusterOperator in Degraded state: %w", clusterOperatorName, err)
-		}
-		return ctrl.Result{}, errors.Wrap(err, "failed to determine state of metal3 deployment")
-	}
-	if deploymentState == appsv1.DeploymentReplicaFailure {
-		err = r.updateCOStatus(ReasonDeployTimedOut, "metal3 deployment rollout taking too long", "")
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("unable to put %q ClusterOperator in Degraded state: %w", clusterOperatorName, err)
-		}
-	} else if deploymentState == appsv1.DeploymentAvailable {
-		err = r.updateCOStatus(ReasonSyncing, "metal3 pod running", "starting other metal3 services")
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("unable to put %q ClusterOperator in Progressing state: %w", clusterOperatorName, err)
-		}
-	}
-
 	for _, ensureResource := range []ensureFunc{
+		provisioning.EnsureMetal3Deployment,
```

Comment:
I think we should avoid having to call CheckExistingMetal3Deployment() twice (and potentially getting inconsistent results). My original intention for that was to add the metal3DeploymentSelector to the ProvisioningInfo struct. That's the tidiest way, but an alternative hack would be to put this in the list:

```
func(info *provisioning.ProvisioningInfo) (bool, error) {
    return provisioning.EnsureMetal3Deployment(info, metal3DeploymentSelector)
}
```

Reply:
Yes, I had considered that. CheckExistingMetal3Deployment() is using a different subset of the return values during these 2 calls. I'll tidy it up by adding it to ProvisioningInfo.

Reply:
Did not change the input parameters for EnsureMetal3Deployment() and used the above hack in addition to adding the LabelSelector to the ProvisioningInfo struct.